BenchmarkVCFs/FindSamplesAndBenchmark.wdl

```diff
@@ -211,98 +211,110 @@ struct Match{
+        docker: docker
+        noAddress: false
+        maxRetries: 1
+        continueOnReturnCode: [0,5]
     }
-  }  
-  
-  String tsv_out="~{metrics_basename}.fingerprints_metrics"
-
-  command <<<
-    bash ~{monitoring_script} > /cromwell_root/monitoring.log &
-
-    
-    java -Xmx7g -jar ~{picard_jar} CrosscheckFingerprints \
-      INPUT=~{sep=" INPUT=" input_data} \
-      SECOND_INPUT=~{sep=" SECOND_INPUT=" ground_truth_files} \
-      HAPLOTYPE_MAP=~{haplotype_database} \
-      LOD_THRESHOLD=-5 \
-      OUTPUT=~{tsv_out} \
-      CROSSCHECK_BY=FILE \
-      CROSSCHECK_MODE=CHECK_ALL_OTHERS \
-      TEST_INPUT_READABILITY=false \
-      EXIT_CODE_WHEN_MISMATCH=5
-      
-   >>>
-   output {
-    File crosscheck = tsv_out
-   }
-
-   runtime {
-    preemptible: preemptible_tries
-    memory: "8 GB"
-    disks: "local-disk " + disk_size + " HDD"
-    docker: docker
-    noAddress: false
-    maxRetries: 1
-    continueOnReturnCode: [0,5]
-  }
 }
 
 task PickMatches {
     input {
         File crosscheck_results
+        File ground_truth_local
+        File ground_truth_global
     }
 
     command <<<
+        python <<CODE
+        import pandas as pd
 
-        cat <<- 'AWK' > prog.awk
-        BEGIN{
-            OFS="	";
-            parsedhead=0;
-        }
-        /^#/{next} 
-        /^$/{next} 
-        parsedhead==0 {
-            parsedhead=1;
-            for (i=1; i<=NF; i++) {
-                f[$i] = i
-            };
-            next;
-            } 
-        $(f["RESULT"])!~/MISMATCH/ && $(f["RESULT"])~/MATCH/ {
-            print $(f["LEFT_FILE"]), $(f["LEFT_SAMPLE"]), $(f["RIGHT_FILE"]), $(f["RIGHT_SAMPLE"])
-            } 
-        AWK
-
-
-        awk -F'\t' -f prog.awk ~{crosscheck_results} > matches.tsv
+        # Import crosscheck results into DataFrame
+        crosscheck = pd.read_csv("~{crosscheck_results}", sep='\t', skiprows=6)
+
+        # Find the pairs which were (expected/unexpected) matches
+        match_list = crosscheck[-crosscheck['RESULT'].str.contains('MISMATCH')]
+
+        # Fix file path back to cloud rather than localized version
+        truth_files_local = pd.read_csv("~{ground_truth_local}")
+        truth_files = pd.read_csv("~{ground_truth_global}")
+        truth_files['LOCAL'] = "file://" + truth_files_local   # Fixes some formatting issue from the crosscheck output
+        local_to_global = dict(zip(truth_files.LOCAL, truth_files.GLOBAL))
+        match_list['RIGHT_FILE'] = match_list['RIGHT_FILE'].map(local_to_global)
+
+        # Write out the left/right files/samples from the list of matches
+        match_list[['LEFT_FILE', 'LEFT_SAMPLE', 'RIGHT_FILE', 'RIGHT_SAMPLE']].to_csv("matches.tsv", sep='\t', index=False, header=False)
+
+        CODE
     >>>
+
     output {
         Array[Array[String]] matches = read_tsv("matches.tsv")
     }
+
     runtime {
-        docker: "ubuntu"
+        docker: "us.gcr.io/broad-dsde-methods/rm-python-data-slim"
```

Comment:
Is there a more "official" docker image than a personalized one? This is probably the most efficient docker image, but it's good to either point to a well-maintained docker image (with the Dockerfile publicly available) or include the Dockerfile in this repo if we want to have a custom one

Reply:
Usually I don't think we optimize for docker image size that much, so if there's a public one that's larger in size then that's fine

Reply:
Unfortunately I haven't been able to find even an "official" looking docker just with pandas installed! I tried looking for generic "data" dockers that would include those libraries as well, but was surprised to not find anything obvious. If you know of any I'd be happy to substitute it.

Reply:
Alternatively I could rename it so it looks less "personalized." The dockerfile exists here in palantir-workflows, so it could just be a docker image we "officially" maintain, though I agree I don't like the idea of using something so generic that it should be standard but doesn't seem to be provided by anyone closer to the developers as far as I could tell.

Reply:
Oh awesome, I didn't realize the Dockerfile was in this repo already, so yeah, we should rename it to have a name without the rm, and perhaps put it into a requester pays bucket, so that we don't get charged every time someone pulls the docker

Reply:
Did the renaming in the latest commit. I'm not sure how to turn a docker image to requester pays. It might be possible if it's stored in a bucket somewhere, by setting it for the bucket, but I haven't been able to hunt down where it's stored yet.